src/lsm/lsm_cursor_bulk.c

```diff
@@ -90,17 +90,29 @@ __clsm_insert_bulk(WT_CURSOR *cursor)
 int
 __wt_clsm_open_bulk(WT_CURSOR_LSM *clsm, const char *cfg[])
 {
+	WT_CONFIG_ITEM cval;
 	WT_CURSOR *cursor, *bulk_cursor;
 	WT_LSM_TREE *lsm_tree;
 	WT_SESSION_IMPL *session;
+	int ordered;
 
 	bulk_cursor = NULL;
 	cursor = &clsm->iface;
 	lsm_tree = clsm->lsm_tree;
+	ordered = 1;	/* Default to ordered inserts */
 	session = (WT_SESSION_IMPL *)clsm->iface.session;
 
 	F_SET(clsm, WT_CLSM_BULK);
 
+	/*
+	 * Check for the undocumented unordered bulk flag, which is used when
+	 * doing index builds into LSM for existing trees.
+	 */
+	WT_RET(__wt_config_gets_def(session, cfg, "bulk", 0, &cval));
+	WT_ASSERT(session, cval.val != 0);
+	if (strncmp(cval.str, "unordered", strlen("unordered")) == 0)
+		ordered = 0;
+
```

Comment:
@michaelcahill I did think about that, and it generalizes the solution. I decided to keep it within LSM for now - since I figured an un-ordered bulk load into a btree muddies the ```bulk``` semantic (though probably less than having a special option that is only supported by LSM).